site/app/libraries/database/DatabaseQueries.php

```diff
@@ -940,6 +940,14 @@ public function insertNewRotatingSection($section) {
         $this->course_db->query("INSERT INTO sections_rotating (sections_rotating_id) VALUES(?)", array($section));
     }
 
+    public function insertNewRegistrationSection($section) {
+        $this->course_db->query("INSERT INTO sections_registration (sections_registration_id) VALUES(?)", array($section));
+    }
+
+    public function deleteRegistrationSection($section) {
+        $this->course_db->query("DELETE FROM sections_registration WHERE sections_registration_id=?", array($section));
```

Comment:
This column is [referenced by another table](https://github.com/Submitty/Submitty/blob/871cc96ea5dc1f952e2cda514cdc619412345f6f/site/data/course_tables.sql#L854-L855).  Although not specifically defined in `course_tables.sql`, a deletion here could trigger a cascade delete on the `grading_registration` table.  Likely, this delete query could entirely fail, if the cascade delete cannot be resolved for some reason.  e.g. `grading_registration` has a multi-column primary key, part of which is referencing `sections_registration`.

Reply:
Hi @pbailie ,
It is ensured that the registration section which is to be deleted do not have any user enrolled in it.
So if we are deleting a registration section (lets say 10) satisfying above condition, then there will be no entry in grading_registration table with 10 since no user will have 10 as registration section. So on deleting 10 in sections_registration table, the grading_registration table same.
Will the above function still cause problem.

Reply:
There's one more check we should add.  Graders can be assigned to one or more registration sections for grading.  We need to make sure that in addition to no users being registered to the registration section, no users/graders are assigned to grade that registration section.

Reply:
ok got it. For now I have shifted the whole thing to SetUp page

![1](https://user-images.githubusercontent.com/21223046/40145384-2d2a60b2-597f-11e8-80e3-33739f571099.png)

![2](https://user-images.githubusercontent.com/21223046/40145391-33d8fa4a-597f-11e8-8b62-39fbec4a6f3b.png)